ddtrace/tracer/span.go

```diff
@@ -74,6 +74,10 @@ func (s *span) BaggageItem(key string) string {
 func (s *span) SetTag(key string, value interface{}) {
 	s.Lock()
 	defer s.Unlock()
+	s.setTagLocked(key, value)
```

Comment:
This could have been removed, but let's keep it - it makes it very clear that the span is locked when it's being modified.

Reply:
Your're right. Let's remove it. It's not used anywhere else, which could be misleading.